specs/commandline/plugin.md

```diff
@@ -41,51 +42,145 @@ Aliases:
 ### notation plugin install
 
 ```text
-Installs a plugin
+Install a plugin
 
 Usage:
-  notation plugin install [flags] <plugin path>
+  notation plugin install [flags] <plugin_source>
 
 Flags:
   -h, --help                    help for install
   -f, --force                   force the installation of a plugin
-
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the location of plugin installation file, options: "file", "url","registry" (default "file")                  
 Aliases:
   install, add
 ```
 
-### notation plugin remove
+### notation plugin upgrade
 
 ```text
-Removes a plugin
+Upgrade a plugin
 
 Usage:
-  notation plugin remove [flags] <plugin name>
+  notation plugin upgrade [flags] <plugin_source>
 
 Flags:
-  -h, --help          help for remove
+  -h, --help                    help for install
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the location of plugin installation file, options: "file", "url","registry" (default "file")                  
```

Comment:
@yizha1  In general, a feature spec can have features in the roadmap or future milestones. The Notary Project spec also has some points and feature requirements that are planned for future releases. 
If we have a feature defined in the spec earlier, it is possible to be implemented anytime. I think we can mark this kind of features as "for future milestone" in spec to avoid confusion.

Reply:
My point is that this section will be used for implementation, however, we won't support registry, so we cannot display it to users, right?